#!/usr/bin/env python
PACKAGE = "morus_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("aut_ref",      bool_t,    0, "Automated reference setting for capturing data",               False)

gen.add("K_I_MPC",         double_t, 0, "Integral gain for the angle",                           5.0, 0.0,   20.0)

# x - states [p0, v0, p1, v1, omega0, omega1, angle, angular_velocity]
gen.add("q_p0",               double_t, 0, "Penalty for moving mass 0 position",                 0.0, 0.0,   10.0)
gen.add("q_v0",               double_t, 0, "Penalty for moving mass 0 velocity",                 0.0, 0.0,   10.0)
gen.add("q_p1",               double_t, 0, "Penalty for moving mass 1 position",                 0.0, 0.0,   10.0)
gen.add("q_v1",               double_t, 0, "Penalty for moving mass 1 velocity",                 0.0, 0.0,   10.0)

gen.add("q_rotor_speed_0",    double_t, 0, "Penalty for rotor 0",                                0.0, 0.0,   10.0)
gen.add("q_rotor_speed_1",    double_t, 0, "Penalty for rotor 1",                                0.0, 0.0,   10.0)

gen.add("q_angle",            double_t, 0, "Penalty for angle of UAV",                        1200.0, 0.0, 2000.0)
gen.add("q_angular_velocity", double_t, 0, "Penalty for angular velocity of UAV",                0.0, 0.0,   10.0)

# u - inputs [p0_ref, p1_ref, omega0_ref, omega1_ref]
gen.add("r_mass_0",           double_t, 0, "Penalty for 0. mass reference position",             0.1, 0.0,   10.0)
gen.add("r_mass_1",           double_t, 0, "Penalty for 1. mass reference position",             0.1, 0.0,   10.0)
gen.add("r_rotor_0",          double_t, 0, "Penalty for 0. rotor reference rotation change",  0.0008, 0.0,   10.0)
gen.add("r_rotor_1",          double_t, 0, "Penalty for 1. rotor reference rotation change",  0.0008, 0.0,   10.0)

# du - change in inputs [p0_ref, p1_ref, omega0_ref, omega1_ref]
gen.add("r_delta_mass_0",     double_t, 0, "Penalty for 0. mass change of reference position",   0.7, 0.0,   10.0)
gen.add("r_delta_mass_1",     double_t, 0, "Penalty for 1. mass change of reference position",   0.7, 0.0,   10.0)
gen.add("r_delta_rotor_0",    double_t, 0, "Penalty for 0. rotor reference rotation change",     0.7, 0.0,   10.0)
gen.add("r_delta_rotor_1",    double_t, 0, "Penalty for 0. rotor reference rotation change",     0.7, 0.0,   10.0)

exit(gen.generate(PACKAGE, "Config", "MPCAttitudeController"))
